watch_test.go

```diff
@@ -72,10 +73,46 @@ func TestWatch(t *testing.T) {
 	assert.Equal(t, data, event.Data.(string), "event data doesn't match")
 }
 
+func TestWatchError(t *testing.T) {
+	t.Parallel()
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		flusher, ok := w.(http.Flusher)
+
+		if !ok {
+			t.Fatal("Streaming unsupported!")
+		}
+
+		w.Header().Set("Content-Type", "text/event-stream")
+		w.Header().Set("Cache-Control", "no-cache")
+		w.Header().Set("Connection", "keep-alive")
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		flusher.Flush()
+		time.Sleep(500 * time.Millisecond)
```

Comment:
What I had in mind would actually work since `httptest.Server` waits for every connection to die before it actually closes.